spring-data-jpa/src/main/java/org/springframework/data/jpa/repository/support/FetchableFluentQueryByPredicate.java

```diff
@@ -175,8 +174,13 @@ public boolean exists() {
 
 	private Page<R> readPage(Pageable pageable) {
 
-		AbstractJPAQuery<?, ?> pagedQuery = pagedFinder.apply(sort, pageable);
-		List<R> paginatedResults = convert(pagedQuery.fetch());
+		AbstractJPAQuery<?, ?> query = pagedFinder.apply(sort, pageable);
+
+		if (!properties.isEmpty()) {
```

Comment:
Do we need to do the same for Query by Example and Specifications?

Reply:
See https://github.com/spring-projects/spring-data-jpa/issues/2721

Reply:
To be more precise, `FetchableFluentQueryByExample` 's `readPage` method taps the `createSortedAndProjectedQuery` method, which applies the expected hint. The same can be said for the `Specification`-based variant. This one is the only one that doesn't do that, probably due to the nature of Querydsl's APIs.